### PR TITLE
BPKButton

### PR DESCRIPTION
**Pull request by tevelee**
https://backpack.github.io/components/button/?platform=web

### CODE REVIEW

File: Backpack/Button/Classes/BPKButton.h [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+ */
+IB_DESIGNABLE @interface BPKButton: UIButton
+- (instancetype)initWithFrame:(CGRect)frame NS_UNAVAILABLE;
+
+/**
+ * Create a `BPKButton` with a specific BPKFont style.
+ * 
+ * @param size Size to be used by the button.
+ * @param style Style of the button.
+ * @see BPKFontStyle
```
Comment (k0nserv) [inline Backpack/Button/Classes/BPKButton.h]: This sohuld be `@see BPKButtonSize` right?

File: Backpack/Button/Classes/BPKButton.m [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+#import <Backpack/BPKGradientView.h>
+#import <Backpack/BPKGradient.h>
+
+NS_ASSUME_NONNULL_BEGIN
+@interface BPKButton()
+
+@end
+
+@implementation BPKButton {
+    BPKGradientView *_backgroundView;
```
Comment (k0nserv) [inline Backpack/Button/Classes/BPKButton.m]: Can we use a property instead of an ivar?

File: Backpack/Button/Classes/BPKButton.m [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+
+- (void)setupWithSize:(BPKButtonSize)size style:(BPKButtonStyle)style imagePosition:(BPKButtonImagePosition)imagePosition {
+    [self setClipsToBounds:YES];
+    [self setAdjustsImageWhenHighlighted:NO];
+    [self setAdjustsImageWhenDisabled:NO];
+    [self.titleLabel setTextAlignment:NSTextAlignmentCenter];
+    
+    _backgroundView = [[BPKGradientView alloc] initWithGradient:[self gradientWithTopColor:BPKColor.clear bottomColor:BPKColor.clear]];
+    [_backgroundView setUserInteractionEnabled:NO];
+    [self insertSubview:_backgroundView atIndex:0];
```
Comment (k0nserv) [inline Backpack/Button/Classes/BPKButton.m]: Hmm adding subViews directly to `UIButton` makes me a bit scared. Maybe we should use the gradient layer here instead

File: Backpack/Button/README.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+```
+pod 'Backpack/Label'
+```
+
+and then run `pod install`.
+
+## Usage
+
+`Backpack/Label` contains the Backpack Label component in the class `BPKLabel` which is a subclass of `UILabel` with Skyscanner styles. It accepts a Backpack font style to set the desired size.
```
Comment (k0nserv) [inline Backpack/Button/README.md]: This still has some copy-and-paste code from Label

File: Example/Backpack.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -395,7 +405,7 @@
 				ORGANIZATIONNAME = Skyscanner;
 				TargetAttributes = {
 					6003F589195388D20070C39A = {
-						DevelopmentTeam = WZLXWEEH2U;
+						DevelopmentTeam = U9XKRL39B6;
```
Comment (k0nserv) [inline Example/Backpack.xcodeproj/project.pbxproj]: This shouldn't have changed?

File: Example/Backpack/View Controllers/BPKButtonSelecorViewController.h [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+//
+//  Created by Laszlo Teveli on 2018. 09. 14..
+//  Copyright © 2018. Skyscanner. All rights reserved.
+//
+
+#import <UIKit/UIKit.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface BPKButtonSelecorViewController : UITableViewController
```
Comment (k0nserv) [inline Example/Backpack/View Controllers/BPKButtonSelecorViewController.h]: This should be `BPKButtonSelectorViewController`

File: Backpack/Button/Classes/BPKButton.h [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+    BPKButtonStyleSecondary = 1,
+    BPKButtonStyleDestructive = 2,
+    BPKButtonStyleFeatured = 3,
+};
+
+/**
+ * Enum values for specifying image position compared to the title label
+ */
+typedef NS_ENUM(NSUInteger, BPKButtonImagePosition) {
+    BPKButtonImagePositionRight = 0,
```
Comment (k0nserv) [inline Backpack/Button/Classes/BPKButton.h]: Can we rename these to `leading` and `trailing`?

Comment (tevelee) [reply Backpack/Button/Classes/BPKButton.h]: Sure, good point!

**Comment by k0nserv** [comment]
Thanks a bunch for the contribution @tevelee 🙌

**Pull request merged**